packages/runtime/container-runtime/src/opLifecycle/outbox.ts

```diff
@@ -328,10 +328,9 @@ export class Outbox {
 		}
 
 		const rawBatch = batchManager.popBatch();
-		if (
-			rawBatch.hasReentrantOps === true &&
-			this.params.groupingManager.shouldGroup(rawBatch)
-		) {
+		const shouldGroup =
+			!disableGroupedBatching && this.params.groupingManager.shouldGroup(rawBatch);
+		if (rawBatch.hasReentrantOps === true && shouldGroup) {
```

Comment:
Adding `disableGroupedBatching` to this `if` statement looks like a bug fix, is that right?  Might want to pull that and any other behavior changes into a separate PR for clarity - so this one can be pure refactor with no behavior change.

If you do that, consider adding all the new test coverage too (if it doesn't depend on the test coverage), to "prove" that the refactor doesn't change behavior.

Reply:
https://github.com/microsoft/FluidFramework/pull/20163